### PR TITLE
WELD-1832 Modify WeldDeployment.loadBeanDeploymentArchive() to follow

### PR DESCRIPTION
**Pull request by mkouba**
the API contract

### CODE REVIEW

**Comment by weld-tester** [comment]
Triggering build using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 396 is now running using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396

**Comment by weld-tester** [comment]
Build 396 outcome was SUCCESS using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/396

File: environments/common/src/main/java/org/jboss/weld/environment/deployment/WeldDeployment.java [hunk 1]
```diff
@@ -64,4 +87,41 @@ public BeanDeploymentArchive getBeanDeploymentArchive(Class<?> beanClass) {
...
+     * @param beanClass
+     * @return the additional bean deployment archive
+     */
+    protected WeldBeanDeploymentArchive createAdditionalBeanDeploymentArchive(Class<?> beanClass) {
+        // At this time only the initial bean class is known
+        Set<String> beanClasses = new HashSet<String>();
+        beanClasses.add(beanClass.getName());
+
+        WeldBeanDeploymentArchive additionalBda = new WeldBeanDeploymentArchive(beanClass.getClassLoader().getClass().getName() + "@"
+                + System.identityHashCode(beanClass.getClassLoader()) + ADDITIONAL_BDA_ID_SUFFIX, beanClasses, null);
```
Comment (jharting) [inline environments/common/src/main/java/org/jboss/weld/environment/deployment/WeldDeployment.java]: System.identityHashCode(beanClass.getClassLoader()) will cause non-deterministict BDA IDs. Can we make these deterministic?

Comment (mkouba) [inline environments/common/src/main/java/org/jboss/weld/environment/deployment/WeldDeployment.java]: Give me a hint :-) `toString()` is not usable and `ClassLoader` does not have any deterministic id...

Comment (jharting) [inline environments/common/src/main/java/org/jboss/weld/environment/deployment/WeldDeployment.java]: No idea yet

**Comment by weld-tester** [comment]
Build 396 outcome was SUCCESS using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/396

**Comment by weld-tester** [comment]
Build 396 outcome was SUCCESS using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/396

**Comment by weld-tester** [comment]
Build 396 outcome was SUCCESS using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/396

**Comment by weld-tester** [comment]
Build 396 outcome was SUCCESS using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/396

**Comment by weld-tester** [comment]
Build 396 outcome was SUCCESS using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/396

**Comment by weld-tester** [comment]
Build 396 outcome was SUCCESS using a merge of cd18cc5cbdcc740c7e1e1849c8ad2e9c9f8e6041 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/396
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/396

File: environments/common/src/main/java/org/jboss/weld/environment/deployment/WeldDeployment.java (hunk 1 above)
Comment (jharting) [inline environments/common/src/main/java/org/jboss/weld/environment/deployment/WeldDeployment.java]: As discussed previously we should probably switch to a simpler single-additional-BDA solution. That would give us the same behavior and does not introduce the bda ID problem.

**Comment by weld-tester** [comment]
Triggering build using a merge of a8759e718bbb516e5eb9d0c9d4789b8cdb72eb23 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/

**Comment by weld-tester** [comment]
Build 422 is now running using a merge of a8759e718bbb516e5eb9d0c9d4789b8cdb72eb23 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/422

**Comment by weld-tester** [comment]
Build 422 outcome was SUCCESS using a merge of a8759e718bbb516e5eb9d0c9d4789b8cdb72eb23 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/422
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/422

**Comment by weld-tester** [comment]
Build 422 outcome was SUCCESS using a merge of a8759e718bbb516e5eb9d0c9d4789b8cdb72eb23 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/422
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/422

**Comment by weld-tester** [comment]
Build 422 outcome was SUCCESS using a merge of a8759e718bbb516e5eb9d0c9d4789b8cdb72eb23 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/422
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/422

**Comment by weld-tester** [comment]
Build 422 outcome was SUCCESS using a merge of a8759e718bbb516e5eb9d0c9d4789b8cdb72eb23 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/422
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/422

**Comment by weld-tester** [comment]
Build 422 outcome was SUCCESS using a merge of a8759e718bbb516e5eb9d0c9d4789b8cdb72eb23 on branch master:
Private: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/Weld-2.x-pull-player-executor-osprey/422
Public: https://hudson.jboss.org/hudson/job/Weld-2.x-pull-player-executor-osprey/422

**Comment by jharting** [comment]
Merged, thanks!

**Pull request closed without merging**